core/src/impl/Kokkos_SharedAlloc.cpp

```diff
@@ -336,12 +338,12 @@ void SharedAllocationRecord<void, void>::print_host_accessible_records(
                reinterpret_cast<uintptr_t>(r->m_next),
                reinterpret_cast<uintptr_t>(r->m_alloc_ptr), r->m_alloc_size,
                r->use_count(), reinterpret_cast<uintptr_t>(r->m_dealloc),
-               r->m_alloc_ptr->m_label);
+               r->m_alloc_ptr ? r->m_alloc_ptr->m_label : " [NOT ALLOCATED]");
```

Comment:
Do we still need that ternary with the excluding the root?
If so you missed L359, otherwise drop that change.

Reply:
Probably not, but like Daniel said it doesn't hurt to have the safeguard. If the record headers get corrupted, this could be the difference between a segfault and printing something reasonable to help debug.

Reply:
I do not agree.  At least if it segfault you will notice and the code is less complex

Reply:
@masterleinad I don't have a strong feeling either way, want to just take it out?

Reply:
I still prefer avoiding the segfault and printing "[NOT ALLOCATED]". I won't request changes if you take it out, though.

Reply:
Wait no, @dalg24 this is already consistent. Line 359 doesn't need the ternary because the non-detailed path checks ``if (r->m_alloc_ptr)`` on 347, and if that is null it prints "<space> [ 0 + 0 ]". I vote to just keep it how it is

Reply:
> I still prefer avoiding the segfault and printing "[NOT ALLOCATED]". I won't request changes if you take it out, though.

If the intent when not processing the root is that we will never have a "non allocated pointer", then the silently skip `nullptr` or print `"[NOT ALLOCATED]"` makes that we would not realize if `nullptr` are introduced in the linked list.
If they occur in normal operation and it absolutely must be handled (which you don't clearly state) then sure.

Reply:
I ended up going @dalg24 's preferred way.